services/job_subscriber.go

```diff
@@ -86,12 +86,16 @@ func (js *jobSubscriber) Disconnect() {
 func (js *jobSubscriber) OnNewHead(head *models.BlockHeader) {
 	pendingRuns, err := js.store.JobRunsWithStatus(models.RunStatusPendingConfirmations, models.RunStatusInProgress)
 	if err != nil {
-		logger.Error(err.Error())
+		logger.Error("error fetching pending job runs:", err.Error())
 	}
 
 	ibn := head.ToIndexableBlockNumber()
 	for _, jr := range pendingRuns {
-		if err := js.store.RunChannel.Send(jr.ID, jr.Result, ibn); err != nil {
+		jr.Overrides, err = jr.Overrides.Merge(jr.Result)
+		if js.store.Save(&jr); err != nil {
+			logger.Error("error updating job run", err.Error())
```

Comment:
If we fail to save the overrides, do we still want to send the job run through the `RunChannel`? It would be run with different parameters. I would say no because it has financial consequences.

Reply:
No we don't, good catch.